includes/cart/class-edd-cart.php

```diff
@@ -717,11 +717,16 @@ public function display_cart_discount( $echo = false ) {
 	 * @since 2.7
 	 *
 	 * @param int   $download_id Download ID of the item to check.
- 	 * @param array $options
+	 * @param array $options
 	 * @return bool
 	 */
 	public function is_item_in_cart( $download_id = 0, $options = array() ) {
-		$cart = $this->get_contents();
+		$cart      = $this->get_contents();
+		$variation = explode( '_', $download_id );
+		if ( ! empty( $variation[1] ) ) {
+			$download_id         = absint( $variation[0] );
+			$options['price_id'] = absint( $variation[1] );
+		}
```

Comment:
Can you help me understand why this change is needed? Just trying to figure out the circumstances in which a `{download_id}_{price_id}` string would be passed into this function.

Reply:
There are use cases where the product ID/price ID string is passed to the function from the discount data ([product requirements](https://github.com/easydigitaldownloads/easy-digital-downloads/blob/issue/8322/includes/class-edd-discount.php#L1429) here). So my understanding is that the cart check function needs to know not just if the product is in the cart, but also if the specific variation is in the cart?

Reply:
Thank you! I figured there was something -- I just couldn't find it.

I am in two minds about what I'd prefer though.. Part of me is inclined to do the exploding logic at [this point](https://github.com/easydigitaldownloads/easy-digital-downloads/blob/issue/8322/includes/class-edd-discount.php#L1429) instead of inside the function itself.

I'll think on it a bit.

Reply:
My thought with putting it where I did is that I would have to do it just the one time in the one place. Otherwise we need to make sure that any time it's used in the context of checking a discount, we need to potentially include the conversion, and that may extend to add-ons and/or third party code. This felt like it would cover multiple bases, I guess, but I'm willing to change.